tests/codegen/llvm/variable.ll

```diff
@@ -13,11 +13,13 @@ target triple = "bpf-pc-linux"
 @AT_y = dso_local global %"struct map_t.0" zeroinitializer, section ".maps", !dbg !25
 @ringbuf = dso_local global %"struct map_t.1" zeroinitializer, section ".maps", !dbg !27
 @event_loss_counter = dso_local global %"struct map_t.2" zeroinitializer, section ".maps", !dbg !41
+@max_cpu_id = dso_local externally_initialized constant i64 zeroinitializer, section ".rodata", !dbg !48
+@read_map_val_buf = dso_local externally_initialized global [1 x [2 x [16 x i8]]] zeroinitializer, section ".data.read_map_val_buf", !dbg !50
```

Comment:
These global variables aren't used in this script. Also in a number of other test cases.

The condition for "required global variable" seems to be a little off.

Reply:
Ah this is because Visitors.cpp recurses into the map AND expression:
```
void Visitor::visit(AssignMapStatement &assignment)
{
  Visit(*assignment.map);
  Visit(*assignment.expr);
}
```

While codegen LLVM does not recurse into the map under certain circumstances (aka only when map.key_expr is set AND not using an aggregate function):
```
  Map &map = *assignment.map;
  auto scoped_del = accept(assignment.expr);

  if (!expr_) // Some functions do the assignments themselves
    return;
```

This kinda further emphasizes the point I make in the summary that CodegenLLVM and ResourceAnalyser are starting to be very tied together. It may make sense to extract some logic out to a base class or do a 2 pass codegen algorithm.